sub/filter.c

```diff
@@ -0,0 +1,240 @@
+/*
+ * This file is part of mpv.
+ *
+ * mpv is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdlib.h>
+#include <assert.h>
+#include <string.h>
+#include "common/common.h"
+#include "common/msg.h"
+#include "sd.h"
+
+// Filter for removing subtitle additions for deaf or hard-of-hearing (SDH)
+// This is for English, but may in part work for others too.
+// The intention is that it can be on always so may not remove
+// all SDH parts.
+// This is for filtering ASS encoded subtitles
+//
+// The incoming ass line is overwritten in place (starting at text position)
+// If result is empty line the first postion of the ass line will be set to NULL.
+//
+// Future optimization could be to include state so format and n_ignored
+// could be done once.
+//
+// Future enhancement could be to add "- " when a speaker label
+// is removed. Need to check what is used in current file as some
+// use "-" and some nothing at all.
+void filter_SDH(struct sd *sd, char * format, int n_ignored, char * ass, int filter_harder)
```

Comment:
It does say that the incoming line in "ass" is overwritten in place. But yes, it could be done better.
I looked around at other parts of mpv and there were little comments on functions anywhere and even less places where I could see how you prefer to explain parameters. So I did not explain them here.